inc/auth.php

```diff
@@ -479,13 +480,22 @@ function auth_ismanager($user = null, $groups = null, $adminonly = false) {
         $groups = $USERINFO ? (array) $USERINFO['grps'] : array();
     }
 
-    // check superuser match
-    if(auth_isMember($conf['superuser'], $user, $groups)) return true;
-    if($adminonly) return false;
-    // check managers
-    if(auth_isMember($conf['manager'], $user, $groups)) return true;
+    // prefer cached result
+    static $cache = [];
+    $cachekey = 'c-' . $user . '-' . $adminonly . '-' . join(':', $groups);
```

Comment:
While I cannot construct an example where this is a problem, I am a bit worried that in theory it might be possible that cache keys might collide. For example if user `foo-1-bar` is in group `admin`, then user `foo` with group `bar-1-admin` would get the same key (if `$adminonly` is set). What would be the performance impact of simply serializing an array of the three values and using that as key?

Reply:
Good question. I did a quick test for a user with 4000 groups and 500 is_admin() checks:

|                        | Time   |
|------------------------|--------|
| No Caching             | 26.6s  |
| Joined Cache Key       | 0.261s |
| Serialized Cache Key   | 0.286s |
| JSON encoded Cache Key | 0.276s |

So I guess it doesn't really matter what we use. I will put in serializing.